packages/gatsby/src/utils/page-ssr-module/entry.ts

```diff
@@ -101,15 +102,34 @@ export async function getData({
   }
   results.pageContext = page.context
 
+  let searchString = ``
+  if (req?.query) {
+    const maybeQueryString = Object.entries(req.query)
+      .map(([k, v]) => `${k}=${v}`)
+      .join(`&`)
+    if (maybeQueryString) {
+      searchString = `?${maybeQueryString}`
+    }
```

Comment:
I think you can rewrite this into this:
```suggestion
  searchString = new URL('http://localhost:8000/test?q=234').search
```